src/java/za/co/mip/ablduck/utilities/HTMLGenerator.java

```diff
@@ -0,0 +1,376 @@
+                         + "</div>";
+        return classHtml;
+    }
+    
+    private String renderSidebar (SourceJSObject cls) {
+        String sidebar = "";
+        
+        if(!"".equals(cls.ext) || cls.subclasses.size() > 0){
+            
+            sidebar = "<pre class='hierarchy'>"
+                         + renderClassTree(cls)
+                         + renderSubclasses(cls)
+                         + renderAuthor(cls)
+                    + "</pre>";
+        }
+        
+        return sidebar;
+    }
+    
+    private String renderAuthor (SourceJSObject cls) {
+        String author = "";
+        if (!"".equals(cls.author)){
+            author = "<h4>Author</h4>"
+                   + "<div class='dependency'>" + cls.author + "</div>";
+        }
+        
+        return author;
+    }
+    
+    private String renderClassTree (SourceJSObject cls) {
+        String classTree = "";
+                
+        if(cls.superclasses.size() > 0) {
+            classTree = "<h4>Hierarchy</h4>";
+            
+            cls.superclasses.add(cls.name);
+        
+            classTree += renderSuperTree(cls.superclasses, 0);
+        }
+        
+        return classTree;
+    }
+    
+    private String renderSuperTree (List<String> superclasses, Integer i) {
+        if (i == superclasses.size())
+            return "";
+        
+        String tree = "";
+        
+        String firstChild = "";
+        if(i==0){ 
+            firstChild = "first-child";
+        }
+            
+        String name = renderLink((String)superclasses.get(i));
+        if (i == superclasses.size() -1){
+            name = "<strong>" + superclasses.get(i) + "</strong>";
+        }
+             
+        tree += "<div class='subclass " + firstChild + "'>"
+              + name
+              + renderSuperTree(superclasses, i + 1)
+              + "</div>";
+        
+        return tree;
+    }
+    
+    private String renderSubclasses (SourceJSObject cls) {
+        String deps = "";
+        if(cls.subclasses.size() > 0){
+            deps = "<h4>Subclasses</h4>";
+
+            for (String subclass : cls.subclasses) {
+                deps += "<div class='dependency'>" + renderLink(subclass) + "</div>";
+            }
+        }
+        return deps;
+    }
+    
+    private String renderClassComment (SourceJSObject cls) {
+        String classComment = "";
+        
+        if (cls.meta.internal != null) 
+            classComment = renderInternal(cls.meta.internal) + "</br>";
+                
+        if (cls.meta.deprecated != null)
+            classComment = renderDeprecated("class", cls.meta.deprecated) + "</br>" + classComment;
+               
+        classComment += cls.comment;
+        return classComment;
+    }
+    
+    private String renderMemberDetails (SourceJSObject cls, String memberSection, String memberSectionTitle) {
+        
+        String member = "<div class='members-section'>"

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "member". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1488)